backend/server.js

```diff
@@ -17,7 +17,27 @@ app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
 const db = require("./app/models");
-db.sequelize.sync();
+const retryMaxCount = 10;
+const retryInterval = 5;
+dbSyncWithRetry(retryMaxCount, retryInterval, db);
+// simple to understand retry function for connecting to the database
+function dbSyncWithRetry(maxRetries, sleepTime, db) {
+  function sleep(ms) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+
+  async function handleError(err) {
+    if (maxRetries <= 0) {
+      throw err;
+    }
+    console.log(`Could not connect to database, waiting for ${sleepTime} seconds. ${maxRetries - 1} retries left`);
+    await sleep(sleepTime * 1000); // Sleeps for 2 seconds
+
+    return dbSyncWithRetry(maxRetries - 1, sleepTime, db);
+  }
+
+  return db.sequelize.sync().catch(handleError);
```

Comment:
Since you're returning a promise from `dbSyncWithRetry` it might be optimal to move the below routing `app.listen` after dbSync is done.

I.E.

```js
dbSyncWithRetry(retryMaxCount, retryInterval, db).then(() => {
  app.listen(PORT, (err) => {
    if (err) {
        throw new Error(`Could not start server: ${err}`);
    }

    console.log(`Server is running on port ${PORT}.`);
  });
});
`

Reply:
Feedback noted, will be handled separately.